ddl/disttask_flow.go

```diff
@@ -0,0 +1,127 @@
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//	http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package ddl
+
+import (
+	"context"
+	"encoding/json"
+
+	"github.com/pingcap/errors"
+	"github.com/pingcap/tidb/disttask/framework/dispatcher"
+	"github.com/pingcap/tidb/disttask/framework/proto"
+	"github.com/pingcap/tidb/kv"
+	"github.com/pingcap/tidb/meta"
+	"github.com/pingcap/tidb/parser/model"
+)
+
+// FlowHandleLitBackfillType is the task type to handle backfill.
+const FlowHandleLitBackfillType = "flowHandleLitBackfill"
+
+// LitBackfillGlobalTaskMeta is the global task meta for lightning backfill.
+type LitBackfillGlobalTaskMeta struct {
+	Job          model.Job `json:"job"`
+	EleID        int64     `json:"ele_id"`
+	EleTypeKey   []byte    `json:"ele_type_key"`
+	IsMergingIdx bool      `json:"is_merging_idx"`
+}
+
+// LitBackfillSubTaskMeta is the subtask meta for lightning backfill.
+type LitBackfillSubTaskMeta struct {
+	PhysicalTableID int64 `json:"physical_table_id"`
+	IsMergingIdx    bool  `json:"is_merging_idx"`
+}
+
+// LitBackfillSubTaskRollbackMeta is the meta for rolling back a lightning fill.
+type LitBackfillSubTaskRollbackMeta struct {
+	IsMergingIdx bool `json:"is_merging_idx"`
+}
+
+type litBackfillFlowHandle struct {
+	getDDL func() DDL
+}
+
+// NewLitBackfillFlowHandle creates a new litBackfillFlowHandle.
+func NewLitBackfillFlowHandle(getDDL func() DDL) dispatcher.TaskFlowHandle {
+	return &litBackfillFlowHandle{
+		getDDL: getDDL,
+	}
+}
+
+// ProcessNormalFlow processes the normal flow.
+func (h *litBackfillFlowHandle) ProcessNormalFlow(_ dispatcher.Dispatch, gTask *proto.Task) (metas [][]byte, err error) {
+	if gTask.State != proto.TaskStatePending {
+		// This flow has only one step, finish task when it is not pending
+		return nil, nil
+	}
+
+	var globalTaskMeta LitBackfillGlobalTaskMeta
+	if err = json.Unmarshal(gTask.Meta, &globalTaskMeta); err != nil {
+		return nil, err
+	}
+
+	d, ok := h.getDDL().(*ddl)
+	if !ok {
+		return nil, errors.New("The getDDL result should be the type of *ddl")
+	}
+
+	job := &globalTaskMeta.Job
+	var tblInfo *model.TableInfo
+	err = kv.RunInNewTxn(d.ctx, d.store, false, func(ctx context.Context, txn kv.Transaction) error {
+		tblInfo, err = meta.NewMeta(txn).GetTable(job.SchemaID, job.TableID)
+		return err
+	})
+
+	var physicalIDs []int64
+	if tblInfo.Partition != nil {
+		defs := tblInfo.Partition.Definitions
+		physicalIDs = make([]int64, len(defs))
+		for i := range defs {
+			physicalIDs[i] = defs[i].ID
+		}
+	} else {
```

Comment:
It must be the partition table.

Reply:
Why it must be partition table?

Reply:
we don't have plan to support non-partition table